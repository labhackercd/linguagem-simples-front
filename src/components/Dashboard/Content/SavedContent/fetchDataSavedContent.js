import axiosInstance from '../../../../auth/axiosApi'


export default async function fetchDataSavedContentCamara(sessionId) {
   
    //TODO - Change api call and uncomment line at agenciaCamara.js
    const response = await axiosInstance.get(("/saved-contents/?session__id="+sessionId), {
    });
    //console.log(response)

    if(response.status===200){
        return response.data;
    }else{
        return null;
    }

};