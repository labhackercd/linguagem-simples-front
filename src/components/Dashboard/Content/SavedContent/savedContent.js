
import React from 'react';
import { Grid, Typography } from '@material-ui/core';
import Box from '@material-ui/core/Box';
import List from '@material-ui/core/List'
import CircularProgress from '@material-ui/core/CircularProgress';


import InputAdornment from '@material-ui/core/InputAdornment';
import SearchIcon from '@material-ui/icons/Search';
import TextField from '@material-ui/core/TextField';
import fetchDataSavedContentCamara from './fetchDataSavedContent'

import NewsCard from './../AgenciaCamara/newsCard'
import TVCard from './../TVCamara/tvCamaraCard'
import RadioCard from './../RadioCamara/radioCard'


function topBarSavedContent(props){
  return(
    <React.Fragment>
      <Grid item xs={10}>
        <Typography variant="h6" style={{ color: "#007E5A" }}>Mais recentes </Typography>
      </Grid>

      <Grid item xs={2}>
        <TextField
          id="input-search-agencia"
          size="small"
          InputProps={{
            endAdornment: (<InputAdornment position="start"><SearchIcon /></InputAdornment>),
          }}
        />
      </Grid>
    </React.Fragment>
  );
}

export default class SavedContent extends React.Component {

  constructor(props){
    super(props);
    this.state = { 
        savedContent: [], 
        dataLoaded: false
    };
  }

  fetchSessionsList = async term => {
    //try {
      const data = await fetchDataSavedContentCamara(this.props.sessionId);

      this.setState({savedContent:data})
      this.setState({dataLoaded:true});
      //console.log(this.state.savedContent)

    //} catch (error) {
    //    throw error;
    //}
  };

  componentDidMount(){
      this._isMounted = true;

      if(this._isMounted){
          this.fetchSessionsList();
      }
  }

  handleCardRenderType(info){
    switch(info.content_type) {
      case 'news':
        return <NewsCard info={info} sessionId={this.props.sessionId}></NewsCard>
      case 'tv':
        return <TVCard info={info} sessionId={this.props.sessionId}></TVCard>;
      case 'radio':
        return <RadioCard info={info} sessionId={this.props.sessionId}></RadioCard>;
      default:
        return null;
    }
  }

  render(){

    if(!this.state.dataLoaded){
      return (<Box display="flex" justifyContent="center" alignItems="center"><CircularProgress></CircularProgress></Box>)
    }

    if(this.state.savedContent.length < 1){
      return (
        <Box width={1} height={1}>
          <Box display="flex" justifyContent="center" alignItems="center" width={"100%"} height={"100%"}>
            <Typography variant="h5" style={{color: "#C4C4C4"}}>Conte√∫dos salvos</Typography>
          </Box>
        </Box>
      )
    }

    return (
      <div>
        <Grid container>
          {topBarSavedContent()}  
          <Grid item xs={12}>
            <Box paddingTop={3}>
              <List style={{maxHeight: '200px', overflow: 'auto'}}>            
                {this.state.savedContent.map((card) => (
                    <li key={`section-${card.id}`}>
                        <Box my={0.5}>{this.handleCardRenderType(card)}</Box>
                    </li>
                ))}
              </List>
            </Box>
          </Grid>
        </Grid>
      </div>
    )
  }
}