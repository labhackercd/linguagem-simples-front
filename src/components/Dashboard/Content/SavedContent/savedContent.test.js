
import React from 'react';
import {shallow} from "enzyme/build";


import { mount} from 'enzyme';
import MockAdapter from "axios-mock-adapter"
import axiosInstance from './../../../../auth/axiosApi'
import CircularProgress from '@material-ui/core/CircularProgress';

import SavedContent from './savedContent'

it("Should render the SavedContentComponent", () => {
    const component = shallow(<SavedContent sessionId={1}/>);

    expect(component.exists()).toEqual(true);
    expect(component).toMatchSnapshot();
});




describe('Testing lifeclycle of SavedContentComponent content', () => {
    const savedContentMockData = [
        {
          "id": 2,
          "created": "2020-09-03T14:25:35.161495-03:00",
          "content_type": "news",
          "title": "Especialistas criticam permissão para rede social pedir documento de identidade do usuário",
          "url": "http://www.camara.leg.br/noticias/619950-ESPECIALISTAS-CRITICAM-PERMISSAO-PARA-REDE-SOCIAL-PEDIR-DOCUMENTO-DE-IDENTIDADE-DO-USUARIO",
          "session": 1
        },
        {
          "id": 3,
          "created": "2020-09-03T14:59:36.657775-03:00",
          "content_type": "tv",
          "title": "TV Câmara ",
          "url": "http://www.camara.leg.br/radio/radioagencia/560001-SINAL-DA-RADIO-CAMARA-CHEGA-A-MANAUS",
          "session": 1
        },
        {
          "id": 4,
          "created": "2020-09-03T15:14:03.465319-03:00",
          "content_type": "radio",
          "title": "Sinal da Rádio Câmara chega a Manaus",
          "url": "http://www.camara.leg.br/radio/radioagencia/560001-SINAL-DA-RADIO-CAMARA-CHEGA-A-MANAUS",
          "session": 1
        },
        {
            "id": 5,
            "created": "2020-09-03T15:14:03.465319-03:00",
            "content_type": "",
            "title": "Null item for test",
            "url": "http://www.camara.leg.br/null",
            "session": 1
        }
    ]

    test("Test when data is not loaded if spin is in the screen", async (done) => {
        // Return a fixed timestamp when moment().format() is called
        const dashboardSessionId = 1;
          
        const  wrapper = shallow(<SavedContent sessionId={dashboardSessionId}/>);
        const containsSpinner = wrapper.containsMatchingElement(<CircularProgress />);
        expect(containsSpinner).toBeTruthy()
        
        //const button = wrapper.find("#saveButtonRadio560001").at(0);
        //console.log(button.debug())
        //button.simulate('click')
        done();
    });

    test("Test when data is loaded if spin is not at the screen", async (done) => {
        // Return a fixed timestamp when moment().format() is called
        const dashboardSessionId = 1;
        var mockInstance = new MockAdapter(axiosInstance);
        await mockInstance.onGet('saved-contents/?session__id=1').reply(200,savedContentMockData)
          
        const wrapper = mount(<SavedContent sessionId={dashboardSessionId}/>);
        

        setImmediate(() => {
            wrapper.update();
            //console.log(wrapper.debug())
            const containsSpinner = wrapper.containsMatchingElement(<CircularProgress />);
            expect(containsSpinner).not.toBeTruthy()
            done();
        })


        //const containsSpinner = wrapper.containsMatchingElement(<CircularProgress />);
        //expect(containsSpinner).toBeTruthy()
        //const button = wrapper.find("#saveButtonRadio560001").at(0);
        //console.log(button.debug())
        //button.simulate('click')
        //done();
    });

    afterAll(() => {
       // mockInstance.restore();
    });


});