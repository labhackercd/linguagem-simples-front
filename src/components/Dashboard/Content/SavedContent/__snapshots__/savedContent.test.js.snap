// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the SavedContentComponent 1`] = `
<Styled(MuiBox)
  alignItems="center"
  display="flex"
  justifyContent="center"
>
  <WithStyles(ForwardRef(CircularProgress)) />
</Styled(MuiBox)>
`;
